diff -ur Eclipse_7.0_49/configure Eclipse_7.0_49.tclpath/configure
--- Eclipse_7.0_49/configure	2019-09-10 13:32:38.000000000 +0200
+++ Eclipse_7.0_49.tclpath/configure	2019-09-18 13:44:08.970953998 +0200
@@ -8843,7 +8843,7 @@
 
 if test "$tcl_requested" = "yes"; then
 
-    TCL_REQUIRED="8.5 8.4 8.3"
+    TCL_REQUIRED="8.6 8.5 8.4 8.3"
 
     if test "$OSCLASS" != "win" ; then
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for X" >&5
@@ -9074,14 +9074,20 @@
 	    fi
 	else
 	    TCL_DIRS="/usr /usr/local"
+	    TCL_LIB_DIRS="lib lib64"
 	    for TCL_DIR in $TCL_DIRS; do
-		if test -d "$TCL_DIR/lib/tcl$TCL_VERSION" ; then
-		    found_tcl=yes
-		    TCL_ARCH_DIR="$TCL_DIR"
-		    TCL_INCL_DIR="$TCL_DIR/include/tcl$TCL_VERSION"
-		    GRAPHLIBS="-L$TCL_ARCH_DIR/lib -ltcl$tcl_lib_version -ltk$tcl_lib_version $GRAPHLIBS"
+		for TCL_LIB_DIR in $TCL_LIB_DIRS; do
+		    if test -d "$TCL_DIR/$TCL_LIB_DIR/tcl$TCL_VERSION" ; then
+			found_tcl=yes
+			TCL_ARCH_DIR="$TCL_DIR"
+		    	TCL_INCL_DIR="$TCL_DIR/include/tcl$TCL_VERSION"
+		    	GRAPHLIBS="-L$TCL_ARCH_DIR/lib -ltcl$tcl_lib_version -ltk$tcl_lib_version $GRAPHLIBS"
+		    	break
+		    fi
+		done
+		if test -n "$found_tcl"; then
 		    break
-		fi
+	    	fi
 	    done
 	    if test -n "$found_tcl"; then
 		break
Only in Eclipse_7.0_49.tclpath: configure.orig
diff -ur Eclipse_7.0_49/lib_tcl/eclipse_arch.tcl Eclipse_7.0_49.tclpath/lib_tcl/eclipse_arch.tcl
--- Eclipse_7.0_49/lib_tcl/eclipse_arch.tcl	2019-09-10 13:32:38.000000000 +0200
+++ Eclipse_7.0_49.tclpath/lib_tcl/eclipse_arch.tcl	2019-09-19 19:39:36.888565207 +0200
@@ -25,64 +25,3 @@
 # compute the ECLiPSe architecture name using Tcl primitives
 #
 
-proc ec_arch {} {
-    global tcl_platform tcl_version
-    switch -glob $tcl_platform(os) {
-	Windows* {
-	    if { $tcl_version >= 8.5} {
-		# use pointerSize if possible - wordSize returns 4
-		if { $tcl_platform(pointerSize) == 8 } { return x86_64_nt }
-	    } elseif { $tcl_platform(wordSize) == 8 } { return x86_64_nt }
-
-	    return i386_nt
-	}
-	SunOS {
-	    switch -glob $tcl_platform(osVersion) {
-		4.*	{ return sun4 }
-		5.* {
-		    switch -glob $tcl_platform(machine) {
-			sun4*	{ return sparc_sunos5 }
-			i86pc	{
-			    # This requires tcl8.4 or later:
-			    switch -glob $tcl_platform(wordSize) {
-				4 { return i386_sunos5 }
-				8 { return x86_64_sunos5 }
-			    }
-			}
-		    }
-		}
-	    }
-	}
-	Linux {
-	    switch -glob $tcl_platform(machine) {
-		alpha	{ return alpha_linux }
-		armv7*	{ return armv7_linux }
-		x86_64	{ 
-		    switch -glob $tcl_platform(wordSize) {
-			4 { return i386_linux }
-			8 { return x86_64_linux }
-		    }
-		}
-		i?86	{ return i386_linux }
-	    }
-	}
-	Darwin {
-	    switch -glob $tcl_platform(machine) {
-		Power*	{ return ppc_macosx }
-		i?86	{ 
-		    # This requires tcl8.4 or later:
-		    switch -glob $tcl_platform(wordSize) {
-			4 { return i386_macosx }
-			8 { # 32 bit kernel
-			    return x86_64_macosx 
-			} 
-		    }
-		}
-		x86_64  { # 64 bit kernel
-		    return x86_64_macosx 
-}
-	    }
-	}
-    }
-    error "Platform $tcl_platform(os) $tcl_platform(osVersion) ($tcl_platform(machine)) not supported"
-}
\ No newline at end of file
diff -ur Eclipse_7.0_49/lib_tcl/eclipse.tcl Eclipse_7.0_49.tclpath/lib_tcl/eclipse.tcl
--- Eclipse_7.0_49/lib_tcl/eclipse.tcl	2019-09-10 13:32:38.000000000 +0200
+++ Eclipse_7.0_49.tclpath/lib_tcl/eclipse.tcl	2019-09-19 19:40:49.476629365 +0200
@@ -479,10 +479,9 @@
 #----------------------------------------------------------------------
 
 set eclipsedir [file dirname [file dirname [info script]]]
-source [file join $eclipsedir lib_tcl eclipse_arch.tcl]
 
 set prev [pwd]
-cd [file join $eclipsedir lib [ec_arch]]
+cd [file join $eclipsedir lib ]
 if { [catch {
 	load [file join . tkexdr[info sharedlibextension]]
 	load [file join . tkeclipse[info sharedlibextension]]
diff -ur Eclipse_7.0_49/lib_tcl/tkec_remote.tcl Eclipse_7.0_49.tclpath/lib_tcl/tkec_remote.tcl
--- Eclipse_7.0_49/lib_tcl/tkec_remote.tcl	2019-09-10 13:32:38.000000000 +0200
+++ Eclipse_7.0_49.tclpath/lib_tcl/tkec_remote.tcl	2019-09-19 19:40:58.731765048 +0200
@@ -738,8 +738,7 @@
 #----------------------------------------------------------------------
 
 set eclipsedir [file dirname [file dirname [info script]]]
-source [file join $eclipsedir lib_tcl eclipse_arch.tcl]
-load [file join $eclipsedir lib [ec_arch] tkexdr[info sharedlibextension]]
+load [file join $eclipsedir lib tkexdr[info sharedlibextension]]
 
 
 # obsolete, for compatibility only
diff -ur Eclipse_7.0_49/RUNME Eclipse_7.0_49.tclpath/RUNME
--- Eclipse_7.0_49/RUNME	2019-09-10 13:32:38.000000000 +0200
+++ Eclipse_7.0_49.tclpath/RUNME	2019-09-18 10:35:08.069808594 +0200
@@ -175,7 +175,7 @@
 
 	# Look for (shared/static) libraries
 
-	tcl_dirs="/lib /usr/lib /usr/local/lib"
+	tcl_dirs="/lib /usr/lib /usr/local/lib /lib64 /usr/lib64 /usr/local/lib64"
 	tcl_version_nodot=`echo $TCL_VERSION|tr -d .`
 	tcl_libs="libtcl$TCL_VERSION.$OBJECTS_SUFFIX libtcl$TCL_VERSION.a\
 		tcl$TCL_VERSION.OBJECTS_SUFFIX\ tcl$tcl_version_nodot.OBJECTS_SUFFIX"
