diff -ur Eclipse_6.2_29/Kernel/src/eclipse_cc.cc Eclipse_6.2_29.header-fixes/Kernel/src/eclipse_cc.cc
--- Eclipse_6.2_29/Kernel/src/eclipse_cc.cc	2015-01-14 02:31:09.000000000 +0100
+++ Eclipse_6.2_29.header-fixes/Kernel/src/eclipse_cc.cc	2019-09-23 16:22:12.464130576 +0200
@@ -76,7 +76,7 @@
 
 	EC_atom() {}
 
-	EC_atom(char * s) { d = ec_did(s,0); }
+	EC_atom(const char * s) { d = ec_did(s,0); }
 
 	EC_atom(dident did)
 	{
Only in Eclipse_6.2_29.header-fixes/Kernel/src: eclipse_cc.cc.orig
diff -ur Eclipse_6.2_29/Kernel/src/eclipseclass.h Eclipse_6.2_29.header-fixes/Kernel/src/eclipseclass.h
--- Eclipse_6.2_29/Kernel/src/eclipseclass.h	2012-02-25 14:36:44.000000000 +0100
+++ Eclipse_6.2_29.header-fixes/Kernel/src/eclipseclass.h	2019-09-23 16:22:12.464130576 +0200
@@ -35,6 +35,7 @@
  *
  */
 
+#pragma once
 
 #ifdef _WIN32
 
Only in Eclipse_6.2_29.header-fixes/Kernel/src: eclipseclass.h.orig
