diff -ur Eclipse_6.2_29/configure Eclipse_6.2_29.tclpath/configure
--- Eclipse_6.2_29/configure	2016-07-24 21:34:43.000000000 +0200
+++ Eclipse_6.2_29.tclpath/configure	2019-09-23 16:20:14.047383790 +0200
@@ -8126,7 +8126,7 @@
 
 if test "$tcl_requested" = "yes"; then
 
-    TCL_REQUIRED="8.5 8.4 8.3"
+    TCL_REQUIRED="8.6 8.5 8.4 8.3"
 
     if test "$OSCLASS" != "win" ; then
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for X" >&5
@@ -8357,14 +8357,20 @@
 	    fi
 	else
 	    TCL_DIRS="/usr /usr/local"
+	    TCL_LIB_DIRS="lib lib64"
 	    for TCL_DIR in $TCL_DIRS; do
-		if test -d "$TCL_DIR/lib/tcl$TCL_VERSION" ; then
-		    found_tcl=yes
-		    TCL_ARCH_DIR="$TCL_DIR"
-		    TCL_INCL_DIR="$TCL_DIR/include/tcl$TCL_VERSION"
-		    GRAPHLIBS="-L$TCL_ARCH_DIR/lib -ltcl$tcl_lib_version -ltk$tcl_lib_version $GRAPHLIBS"
+		for TCL_LIB_DIR in $TCL_LIB_DIRS; do
+		    if test -d "$TCL_DIR/$TCL_LIB_DIR/tcl$TCL_VERSION" ; then
+			found_tcl=yes
+			TCL_ARCH_DIR="$TCL_DIR"
+		    	TCL_INCL_DIR="$TCL_DIR/include/tcl$TCL_VERSION"
+		    	GRAPHLIBS="-L$TCL_ARCH_DIR/lib -ltcl$tcl_lib_version -ltk$tcl_lib_version $GRAPHLIBS"
+		    	break
+		    fi
+		done
+		if test -n "$found_tcl"; then
 		    break
-		fi
+	    	fi
 	    done
 	    if test -n "$found_tcl"; then
 		break
Only in Eclipse_6.2_29.tclpath: configure.orig
Only in Eclipse_6.2_29/lib_tcl: eclipse_arch.tcl
diff -ur Eclipse_6.2_29/lib_tcl/eclipse.tcl Eclipse_6.2_29.tclpath/lib_tcl/eclipse.tcl
--- Eclipse_6.2_29/lib_tcl/eclipse.tcl	2012-02-19 18:54:49.000000000 +0100
+++ Eclipse_6.2_29.tclpath/lib_tcl/eclipse.tcl	2019-09-23 16:20:14.048383804 +0200
@@ -479,10 +479,9 @@
 #----------------------------------------------------------------------
 
 set eclipsedir [file dirname [file dirname [info script]]]
-source [file join $eclipsedir lib_tcl eclipse_arch.tcl]
 
 set prev [pwd]
-cd [file join $eclipsedir lib [ec_arch]]
+cd [file join $eclipsedir lib ]
 if { [catch {
 	load [file join . tkexdr[info sharedlibextension]]
 	load [file join . tkeclipse[info sharedlibextension]]
diff -ur Eclipse_6.2_29/lib_tcl/tkec_remote.tcl Eclipse_6.2_29.tclpath/lib_tcl/tkec_remote.tcl
--- Eclipse_6.2_29/lib_tcl/tkec_remote.tcl	2006-09-23 03:54:19.000000000 +0200
+++ Eclipse_6.2_29.tclpath/lib_tcl/tkec_remote.tcl	2019-09-23 16:20:14.048383804 +0200
@@ -738,8 +738,7 @@
 #----------------------------------------------------------------------
 
 set eclipsedir [file dirname [file dirname [info script]]]
-source [file join $eclipsedir lib_tcl eclipse_arch.tcl]
-load [file join $eclipsedir lib [ec_arch] tkexdr[info sharedlibextension]]
+load [file join $eclipsedir lib tkexdr[info sharedlibextension]]
 
 
 # obsolete, for compatibility only
diff -ur Eclipse_6.2_29/RUNME Eclipse_6.2_29.tclpath/RUNME
--- Eclipse_6.2_29/RUNME	2015-01-14 02:31:07.000000000 +0100
+++ Eclipse_6.2_29.tclpath/RUNME	2019-09-23 16:20:14.048383804 +0200
@@ -175,7 +175,7 @@
 
 	# Look for (shared/static) libraries
 
-	tcl_dirs="/lib /usr/lib /usr/local/lib"
+	tcl_dirs="/lib /usr/lib /usr/local/lib /lib64 /usr/lib64 /usr/local/lib64"
 	tcl_version_nodot=`echo $TCL_VERSION|tr -d .`
 	tcl_libs="libtcl$TCL_VERSION.$OBJECTS_SUFFIX libtcl$TCL_VERSION.a\
 		tcl$TCL_VERSION.OBJECTS_SUFFIX\ tcl$tcl_version_nodot.OBJECTS_SUFFIX"
