diff --git a/Kernel/src/eclipse_cc.cc b/Kernel/src/eclipse_cc.cc
index 22ad9cdc..a4b708c5 100644
--- a/Kernel/src/eclipse_cc.cc
+++ b/Kernel/src/eclipse_cc.cc
@@ -68,7 +68,7 @@ class EC_atom
 
 	EC_atom() {}
 
-	EC_atom(char * s) { d = ec_did(s,0); }
+	EC_atom(const char * s) { d = ec_did(s,0); }
 
 	EC_atom(dident did)
 	{
diff --git a/Kernel/src/eclipseclass.h b/Kernel/src/eclipseclass.h
index 424f50db..bd92df74 100644
--- a/Kernel/src/eclipseclass.h
+++ b/Kernel/src/eclipseclass.h
@@ -35,6 +35,7 @@
  *
  */
 
+#pragma once
 
 #include "eclipse_cc.h"
 
